### PR TITLE
OAT: Update relationship multiplicity logic by version

### PR DESCRIPTION
**Pull request by msnyder-msft**
### Summary of changes 🔍 
Updating the handling of multiplicity attributes based on the version of the DTDL model the relationship is attached to.

Removing upgrade button from property inspector, only will have in the modal
image
Update button on modal to have text
image

### Testing 🧪
 > [ Add any special instructions needed to test or view your changes such as environment URLs or other config details. ]

### Checklist ✔️
- [ ] Linked associated issue (if present)
- [ ] Added relevant labels
- [ ] Requested developer reviews
- [ ] Request UI review from design / PM
- [ ] Verify all code checks are passing

### CODE REVIEW

File: src/Components/OATPropertyEditor/Internal/FormRootModelDetails/Internal/FormRootModelDetailsContent/PropertyDetailsEditorModalContent.tsx [hunk 1]
```diff
@@ -134,22 +147,16 @@ const PropertyDetailsEditorModalContent: React.FC<IModalFormRootModelContentProp
...
+    const onUpgradeVersion = useCallback(() => {
+        onUpdateItem(
+            produce((draft) => {
+                return updateDtdlVersion(
+                    draft as DtdlInterface,
+                    DTDL_CONTEXT_VERSION_3
+                );
+            })
+        );
+    }, [onUpdateItem]);
```
Comment (iremgokce) [inline src/Components/OATPropertyEditor/Internal/FormRootModelDetails/Internal/FormRootModelDetailsContent/PropertyDetailsEditorModalContent.tsx:159]: add updateDtdlVersion to dependency list?

File: src/Components/OATPropertyEditor/Internal/FormRootModelDetails/Internal/FormRootModelDetailsContent/PropertyDetailsEditorModalContent.tsx [hunk 2]
```diff
@@ -656,7 +657,7 @@ const PropertyDetailsEditorModalContent: React.FC<IModalFormRootModelContentProp
                                                     draft
                                                 )
                                             ) {
-                                                // delete draft.minMultiplicity;
+                                                delete draft.minMultiplicity;
                                                 return draft;
```
Comment (iremgokce) [inline src/Components/OATPropertyEditor/Internal/FormRootModelDetails/Internal/FormRootModelDetailsContent/PropertyDetailsEditorModalContent.tsx:661]: not in this PR, but do we need to "return" draft in produce?

File: src/Components/OATPropertyEditor/Internal/PropertiesModelSummary.tsx [hunk 3]
```diff
@@ -219,34 +240,38 @@ export const PropertiesModelSummary: React.FC<IPropertiesModelSummaryProps> = (
...
+                isValidModelVersion(
+                    value.trim(),
+                    getDtdlVersionFromContext(modelContext),
+                    false
+                )
+            ) {
+                setModelVersion(value.trim());
+            }
+        },
+        [modelContext]
```
Comment (iremgokce) [inline src/Components/OATPropertyEditor/Internal/PropertiesModelSummary.tsx:255]: checking dependency list if it is important, isValidModelVersion, getDtdlVersionFromContext, setModelVersion

File: src/Components/OATPropertyEditor/Internal/PropertiesModelSummary.tsx [hunk 4]
```diff
@@ -219,34 +240,38 @@ export const PropertiesModelSummary: React.FC<IPropertiesModelSummaryProps> = (
...
+            if (
+                selectedItem &&
+                isDTDLReference(selectedItem) &&
+                isValidReferenceName(value.trim(), selectedItem['@type'], false)
+            ) {
+                setRelationshipName(value.trim());
+            }
         },
-        [commitModelIdChange, modelPath, modelUniqueName, modelVersion]
+        [selectedItem]
```
Comment (iremgokce) [inline src/Components/OATPropertyEditor/Internal/PropertiesModelSummary.tsx:267]: yeah dependency list check in general to include variables

File: src/Components/OATPropertyEditor/Internal/PropertiesModelSummary.tsx [hunk 5]
```diff
@@ -351,8 +376,30 @@ export const PropertiesModelSummary: React.FC<IPropertiesModelSummaryProps> = (
...
+                            <TextField
+                                aria-labelledby={'oat-model-path'}
+                                onBlur={() => commitModelIdChange(itemId)}
+                                onChange={onChangeVersion}
+                                styles={
+                                    classNames.subComponentStyles.stringField
+                                }
+                                value={modelVersion}
+                            />
+                            {/* <SpinButton
```
Comment (iremgokce) [inline src/Components/OATPropertyEditor/Internal/PropertiesModelSummary.tsx:388]: remove if not used. I see that the other PR is connected to this one, so maybe intentional...

File: src/Components/OATPropertyEditor/Internal/FormRootModelDetails/Internal/FormRootModelDetailsContent/PropertyDetailsEditorModalContent.tsx (hunk 2 above)
Comment (msnyder-msft) [reply src/Components/OATPropertyEditor/Internal/FormRootModelDetails/Internal/FormRootModelDetailsContent/PropertyDetailsEditorModalContent.tsx:661]: I think so, becaues the return of this function is what gets passed to set state. 🤔

File: src/Components/OATPropertyEditor/Internal/FormRootModelDetails/Internal/FormRootModelDetailsContent/PropertyDetailsEditorModalContent.tsx (hunk 1 above)
Comment (msnyder-msft) [reply src/Components/OATPropertyEditor/Internal/FormRootModelDetails/Internal/FormRootModelDetailsContent/PropertyDetailsEditorModalContent.tsx:159]: it's a utility function, not a callback so it doesn't have a reference.

**Pull request merged**